doctors.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+import scrapy
+from scrapy.utils.response import open_in_browser
+import urlparse
+
+
+class DoctorsSpider(scrapy.Spider):
+    name = 'doctors'
+    allowed_domains = ['www.uclahealth.org']
+    url = 'https://www.uclahealth.org/body.cfm?id=10&amp;action=list&fullname_search=true&amp;sortby=name&amp;pageno=1&amp;view=list'
+    site_url = 'https://www.uclahealth.org/'
+    start_urls = [url]
+
+    # CSS selectors
+    doctor_url_selector = 'div.list_lastname > a::attr(href)'
+    doctor_name_selector = '#updateProfile > span::text'
+    doctor_detail_selector = '#div_main_0'
+    page_results_selector = '.PageResults > strong::text'
+
+    # XPATH selectors
+    doctor_tabs_selector = "//li[contains(concat(' ', normalize-space(@class), ' '), ' tab ')]"
+    doctor_tab_name_selector = './a/text()'
+    doctor_about_tab_text_selector = '//div[@id=$val]/dl/dd/p/text()'
+    doctor_other_tabs_text_selector = '//div[@id=$val]/text()'
+    # params
+    name_search_param = 'limit_FullName'
+    name_toggle_param = 'fullname_search'
+    page_no_param = 'pageno'
+
+    count = None
+    current_page = 1
+
+    def parse(self, response):
+        formname = 'form_search'
+        formdata = {
+            self.name_search_param: 'a',
+            self.name_toggle_param: 'true',
+        }
+
+        yield scrapy.FormRequest.from_response(response,
+                                               formname=formname,
+                                               formdata=formdata,
+                                               callback=self.doctors_list)
+
+    def doctors_list(self, response):
+
+        for url in response.css(self.doctor_url_selector).extract():
+            url = urlparse.urljoin(self.site_url, url)
+            yield scrapy.Request(url=url, callback=self.parse_doctors)
+
+        if self.count is None:
+            def ceil_div(a, b):
+                return -(-a // b)
+
+            page_results = response.css(
+                self.page_results_selector).extract_first()
+            page_results = page_results.split(' ')
+
+            self.count = ceil_div(int(page_results[-1]), 10)
```

Comment:
regular expression, write a metod which returns you no of pages after readin a string of pagination